### DISCUSSION TITLE
Secure connection for MINIO

### DISCUSSION THREAD

**Discussion by colinmackinlay** [beyondcode/herd-community#719] [Ideas]
When serving a site from Herd as a secured site there is a difficulty I hav efound that assets I want to include stored in minio cannot be served from the Herd version which runs over http.

Browsers recognise the mixed content and redirect to https which fails:

```
Mixed Content: The page at '<URL>' was loaded over HTTPS, but requested an insecure element '<URL>'. This request was automatically upgraded to HTTPS, For more information see <URL>
Mixed Content: The page at 'https://mydomain.test' was loaded over HTTPS, but requested an insecure element 'http://localhost:9000/herd-bucket/logos/fvtyWO8NOmwb2nyS762k6IogT5DEAO3gdOtNMusu.png'.
```

Is it possible now or in the future to configure MINIO to respond to https traffic?

**Comment by rginnow** [comment]
Having the same issue today. I've seen people say that the Minio route needs to be exposed like a site (see: https://tkacz.pro/laravel-temporary-urls-minio/)

I'd like to be able to easily configure the MinIO service to allow HTTPS, or allow it to be configured with both secure and unsecure endpoints for local development across projects (i.e. `http://localhost:9000` for unsecure, or `https://minio.test` for secure)

For now I'm going to attempt to add a manual cert for it by following their docs here: https://min.io/docs/minio/macos/operations/network-encryption.html#enabling-tls

**Reply by rginnow** [reply]
nevermind! Herd has proxy addresses.

Run:
`herd proxy --secure minio.test http://127.0.0.1:9000`
and
`herd proxy --secure console.minio.test http://localhost:60882` (or whatever port it runs on for you)
to set up your endpoints for the app and for your minio console.

if you get the `mc` command from minio you can then run
`mc alias set local https://minio.test <KEY> <SECRET>` with your key and secret
and then
`mc admin info local`

You should see an output that shows your bucket with the correct address.

**Comment by mpociot** [comment]
Closing this as you can now create a secured MinIO server

**Reply by colinmackinlay** [reply]
Brilliant, thank you @mpociot
